frontend/packages/console-dynamic-plugin-sdk/src/api/useResolvedExtensions.ts

```diff
@@ -43,42 +35,36 @@ import {
  */
 export const useResolvedExtensions = <E extends Extension>(
   ...typeGuards: ExtensionTypeGuard<E>[]
-): [ResolvedExtension<E>[], boolean, any] => {
+): [ResolvedExtension<E>[], boolean, any[]] => {
   const extensions = useExtensions<E>(...typeGuards);
 
   const [resolvedExtensions, setResolvedExtensions] = React.useState<ResolvedExtension<E>[]>([]);
   const [resolved, setResolved] = React.useState<boolean>(false);
-  const [error, setError] = React.useState<any>(undefined);
+  const [errors, setErrors] = React.useState<any[]>([]);
 
   React.useEffect(() => {
     let disposed = false;
 
-    Promise.all(
-      extensions.map(async (e) => {
-        const resolvedProperties = await resolveCodeRefProperties(e);
-        return mergeExtensionProperties(e, resolvedProperties) as ResolvedExtension<E>;
-      }),
-    )
-      .then((result) => {
-        if (!disposed) {
-          setResolvedExtensions(result);
-          setResolved(true);
-        }
-      })
-      .catch((err) => {
-        if (!disposed) {
-          setError(err);
-        }
-      });
+    // eslint-disable-next-line promise/catch-or-return
+    Promise.allSettled(
+      extensions.map((e) => resolveExtension<typeof e, any, ResolvedExtension<E>>(e)),
+    ).then((results) => {
+      if (!disposed) {
+        const [fulfilledValues, rejectedReasons] = unwrapPromiseSettledResults(results);
+        setResolvedExtensions(fulfilledValues);
+        setErrors(rejectedReasons);
```

Comment:
Is there a way of knowing which error was caused by which unresolved extension?

Reply:
@rohitkrai03 Do we have a use case for that, besides potentially logging all of these errors?

If we do, we can adapt the hook result type, e.g. an array of

```ts
{
  resolved: boolean;
  extension?: ResolvedExtension<E>; // if resolved is true
  error?: any; // if resolved is false
}
```

which is basically an analogy of [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) result type.

Reply:
> Is there a way of knowing which error was caused by which unresolved extension?

I think the only association here is the order of objects within `Promise.allSettled` result.

Reply:
I don't see any use case with our current codebase. Was just thinking if there was a way we could use it for better error logging. But the current API of `useResolvedExtensions` seem simpler to me and is already being used in a lot of places. So I won't push for a change in API because of this. We can maybe later do this if there's an actual use case for it.